### PR TITLE
Create DkmCustomUIVisualizerInfo instances with ExtensionPartId field

### PR DESCRIPTION
**Pull request by mpeyrotc**
Debugger Managed Custom Visualizers can now be installed via .VSIX packages into Visual Studio. However, we need a new mechanism to identify them within the new extension manager that processes them. Therefore, visualizers now have a new GUID field for that purpose.

Since the Debugger Component that instantiates visualizers receives its information from the `DkmSuccessEvaluationResult` that the Result Provider from Roslyn provides, we need it to also create the `DkmCustomUIVisualizerInfo` collection that is passes to the `DkmSuccessEvaluationResult` with the new field.

Hence, this change makes sure to create new `DkmCustomUIVisualizerInfo` instances with the new `ExtensionPartId` field.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=65730) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=65730) before we can accept your contribution.<br/><br/>:x: mpeyrotc [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=65730)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn?pullRequest=65730) it.</sub>

**Comment by mpeyrotc** [comment]
@gregg-miskelly @dustincoleman FYI.

File: eng/Versions.props [hunk 1]
```diff
@@ -140,10 +140,10 @@
     <MicrosoftVisualStudioComponentModelHostVersion>17.4.248</MicrosoftVisualStudioComponentModelHostVersion>
     <MicrosoftVisualStudioCompositionVersion>16.9.20</MicrosoftVisualStudioCompositionVersion>
     <MicrosoftVisualStudioCoreUtilityVersion>$(VisualStudioEditorPackagesVersion)</MicrosoftVisualStudioCoreUtilityVersion>
-    <MicrosoftVisualStudioDebuggerUIInterfacesVersion>17.4.0-beta.22368.1</MicrosoftVisualStudioDebuggerUIInterfacesVersion>
+    <MicrosoftVisualStudioDebuggerUIInterfacesVersion>17.5.1120201-preview</MicrosoftVisualStudioDebuggerUIInterfacesVersion>
```
Comment (gregg-miskelly) [inline eng/Versions.props:143]: >MicrosoftVisualStudioDebuggerUIInterfacesVersion [](http://example.com/codeflow?start=5&length=48)

I don't think you need to touch this.

**Comment by gregg-miskelly** [comment]
CC @tmat

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by mpeyrotc** [comment]
@tmat, what is your policy regarding changes in the debug engine reaching VS? The change has made it into the release branch for 17.5 and is also in main. Is there anything else I should wait on before it can be completed?

**Comment by mpeyrotc** [comment]
@sharwell could I bother you with a review?

**Comment by tmat** [comment]
Merging, it's good to go.

**Pull request merged**